src/utils/rateLimiter.util.ts

```diff
@@ -0,0 +1,27 @@
+import { RedisService } from '@liaoliaots/nestjs-redis';
+import { Injectable, Logger } from '@nestjs/common';
+
+@Injectable()
+export class RateLimiterService {
+  private readonly logger = new Logger(RateLimiterService.name);
+
+  constructor(private readonly redisService: RedisService) {}
+
+  async isRateLimited(senderId: string, repoId: string): Promise<boolean> {
+    const client = this.redisService.getClient();
+    const key = `rate_limit:${senderId}:${repoId}`;
+    const currentCount = await client.get(key);
+
+    if (currentCount && parseInt(currentCount) >= 5) {
+      return true;
+    }
+
+    await client
+      .multi()
+      .incr(key)
+      .expire(key, 60) // TTL of 60 seconds
+      .exec();
+
+    return false;
+  }
```

Comment:
PullProofAI

Consider adding error handling to handle potential exceptions during Redis operations.
```diff
+     try {
+         // Redis operations
+     } catch (error) {
+         this.logger.error(`Error in Redis operation: ${error.message}`);
+         throw new Error('Rate limiting operation failed');
+     }
```
This addition ensures that any errors during Redis operations are caught, logged, and the function appropriately handles them.



<!-- This is an auto-generated comment by PullProofAI -->